### PR TITLE
feat(license-service): generating updates pkpass

### PR DESCRIPTION
**Pull request by thorkellmani**
## What

* When generating a pkpass, if there is an existing pass it'll be updated according to the live license data.

## Why

Generating a pkpass should update the pass as well

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/clients/smartsolutions/src/lib/smartSolutions.api.ts [hunk 1]
```diff
@@ -210,49 +210,92 @@ export class SmartSolutionsApi {
...
-        },
+        data: res.data.upsertPass,
       }
     }
 
     return res
   }
 
-  async upsertPkPass(payload: PassDataInput): Promise<Result<Pass>> {
+  private async updateExistingPass(
```
Comment (osk) [inline libs/clients/smartsolutions/src/lib/smartSolutions.api.ts:241]: Shouldn't this also update the `expiryDate` in Smart?

File: libs/clients/smartsolutions/src/lib/smartSolutions.api.ts [hunk 2]
```diff
@@ -210,49 +210,92 @@ export class SmartSolutionsApi {
...
+      //if failure, return the response
+      return getPassRes
+    }
+    const passInputData = mapPassToPassDataInput(getPassRes.data.pass)
+
+    const inputFieldValues = mergeInputFields(
+      passInputData.inputFieldValues ?? undefined,
+      payload.inputFieldValues ?? undefined,
+    )
+    //now we finally have the updated pass data!
```
Comment (osk) [inline libs/clients/smartsolutions/src/lib/smartSolutions.api.ts:264]:
```suggestion
```

it's an easy process now, no need to be dramatic 😎

File: libs/clients/smartsolutions/README.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 # Smart Solutions
 
+A library enabling interaction with of digital licenses.
```
Comment (osk) [inline libs/clients/smartsolutions/README.md:3]: with what of?

**Review by osk — CHANGES_REQUESTED** [review]

File: libs/clients/smartsolutions/README.md (hunk 3 above)
Comment (thorkellmani) [reply libs/clients/smartsolutions/README.md:3]: 😩

File: libs/clients/smartsolutions/src/lib/smartSolutions.api.ts (hunk 1 above)
Comment (thorkellmani) [reply libs/clients/smartsolutions/src/lib/smartSolutions.api.ts:241]: it does, the expiryDate is part of the payload which the client sends over
image

Comment (thorkellmani) [reply libs/clients/smartsolutions/src/lib/smartSolutions.api.ts:241]: We just have to be careful to do the same with other clients, which is on the to-do list

File: libs/clients/smartsolutions/src/lib/smartSolutions.api.ts (hunk 2 above)
Comment (thorkellmani) [reply libs/clients/smartsolutions/src/lib/smartSolutions.api.ts]: It wasn't until it was!

**Pull request merged**